h/static/scripts/root-thread.js

```diff
@@ -119,12 +143,17 @@ function RootThread($rootScope, annotationUI, features, searchFilter, viewFilter
     $rootScope.$on(event, function (event, annotation) {
       var annotations = [].concat(annotation);
 
+      if (event.name === events.BEFORE_ANNOTATION_CREATED) {
```

Comment:
We now have two different conditionals in here that check if `event.name === events.BEFORE_ANNOTATION_CREATED`. I think it would make more sense for this event to have a different handler function at this point.